src/Http.elm

```diff
@@ -443,8 +443,12 @@ handleResponse handle response =
             fail (UnexpectedPayload "Response body is a blob, expecting a string.")
 
   else
-
-      fail (BadResponse response.status response.statusText)
+      let
+        text = case response.value of
+          Text str -> str
+          _ -> response.statusText
```

Comment:
Do we know what type is the `response.value` when there is no body? Is it an empty string? If so, it would defeat the purpose, and always display an empty body instead of the `response.statusText`.

Would a test help prove (or disprove) that? (I see there's no tests for this library at the moment, would that be useful?)